security.tex

```diff
@@ -0,0 +1,251 @@
+
+\subsection{Serial Protocol for Authentication}
+The serial protocol for authentication is simple, taking on the Type, Length, Value (TLV) packet format. This format shall ensure that all messages transmitted over the 32-bit \Rauthdata {\em serial port} are clearly transmitted with adequate opportunities for error correction. 
+
+An authentication packet always takes the following format:
+\begin{center}
+	\begin{tabular}{p{4.0 ex}p{4.0 ex}p{8.0 ex}p{16.0 ex}}
+		{\scriptsize 31} & \multicolumn{1}{r}{\scriptsize 24}
+		&
+		{\scriptsize 23} & \multicolumn{1}{r}{\scriptsize 0} \\
+		\hline
+		\multicolumn{2}{|c|}{$|Type|$} & \multicolumn{2}{c|}{$|Length|$} \\
+		\hline
+		\multicolumn{2}{c}{\scriptsize 8} & \multicolumn{2}{c}{\scriptsize 24} \\
+	\end{tabular}
+\end{center}
+
+\begin{center}
+	\begin{xtabular}{|l|p{0.5\textwidth}|c|l|}
+	\hline
+    Field & Description \\
+	\hline
+	|Type| & The type of message being sent to the AM.\\
+    |Length| & The bytes in the value, in MSB, padded to a multiple of 4.\\
+	\hline
+	\end{xtabular}
+\end{center}
+
+The length of each message is sent in MSB order, and must always be a multiple of 4, since the {\em serial port} is, essentially, a 32-bit register. The padding used to create a message with a multiple of 4 must be an octet of all zero bits (0x00). The length value is right-shifted by two bits, since the bottom two bits of any message length will always be zero. The length value {\em only} indicates the length of the value and does {\em not} include the type/length word, nor the appended CRC32.
+
+The value is sent once an {\bf OK} message is received from the AM. This can be achieved by polling the {\tt authbusy} bit. When zero, the debugger can read the response from \Rauthdata. If the value returned by the AM is anything other than {\bf OK}, the debugger should either close, or try again, depending on the error condition returned.
+
+A simple 32-bit cyclic redundancy check (CRC) is appended to each message. This helps ensure that large messages containing cryptographic data are transmitted correctly across the DTM Transport, the DMI, and the serial port connecting the DM to the AM. The CRC value should be sent in MSB order. The CRC is calculated on both the value {\em and} the type/length header. 
+
+\subsubsection{Transceiving}
+The overall process for sending a message to the AM is as follows:
+
+\paragraph{Message Construction}
+\begin{enumerate}
+    \item Pad the message, if necessary, to a multiple of 4 octets.
+    \item Determine the message type.
+    \item Construct the message type/length header word. 
+    \item Prepend the type/length header to the message.
+    \item Calculate the CRC32 checksum across the header and the message.
+    \item Append the CRC32 checksum to the message.
+\end{enumerate}
+
+\paragraph{Message Transmission}
+For each 32-bit word in the message:
```

Comment:
It seems like it could be pretty inefficient if you have to check for OK after every word. Is that really necessary, other than just checking 'busy'?

One could imagine wanting to stream in a bunch of data, then just checking busy/error at the end (that's sort of the pattern in the rest of the spec)